velox/functions/sparksql/SplitFunctions.cpp

```diff
@@ -134,10 +135,127 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
               .build()};
 }
 
+class StringToMap final : public exec::VectorFunction {
+ public:
+  StringToMap(char entryDelim, char keyValueDelim)
+      : entryDelim_(entryDelim), keyValueDelim_(keyValueDelim) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    DecodedVector* inputString = decodedArgs.at(0);
+
+    BaseVector::ensureWritable(
+        rows, MAP(VARCHAR(), VARCHAR()), context.pool(), result);
+    exec::VectorWriter<Map<Varchar, Varchar>> resultWriter;
+    resultWriter.init(*result->as<MapVector>());
+
+    context.applyToSelectedNoThrow(rows, [&](vector_size_t row) {
+      folly::F14FastSet<StringView> keys;
+      resultWriter.setOffset(row);
+      auto& mapWriter = resultWriter.current();
+
+      const StringView& current = inputString->valueAt<StringView>(row);
+      const char* pos = current.begin();
+      const char* end = pos + current.size();
+      const char* entryEnd;
+      const char* keyEnd;
+      do {
+        entryEnd = std::find(pos, end, entryDelim_);
+        keyEnd = std::find(pos, entryEnd, keyValueDelim_);
+        const auto key = StringView(pos, keyEnd - pos);
+        VELOX_USER_CHECK(
+            keys.insert(key).second,
+            "Duplicated keys ('{}') are not allowed.",
```

Comment:
It is better to put dynamic info at the end of the error message. It allows to search the code for the error message easily.

```
Duplicate keys are not allowed: {}
```